### PR TITLE
Add data names to the info print statement

### LINKED ISSUE #555: Mention data names in convergence and iterations file
Currently, e.g.
`precice-SOLIDZ-iterations.log`:
```
Timesteps  Total_Iterations  Iterations  Convergence  avgConvRate(0)  avgConvRate(1)  avgConvRate(2)  avgConvRate(3)  avgConvRate(4)  deleted_Columns
1  4  4  1  0.0131513617329646  0.0097355863572098  0.0121224497932821  0.0148052718537714  0.0097533280718913  0
2  8  4  1  0.0149890377161203  0.0131392422263492  0.0306217024085128  0.0316058307866920  0.0189617635359644  1
3  13  5  1  0.0083313444673868  0.0090989214732156  0.0335150447116706  0.0433537858697371  0.0314135230849259  0
```
and
`precice-SOLIDZ-convergence.log`:
```
Timestep  Iteration  resNorm(0)  resNorm(1)  resNorm(2)  resNorm(3)  resNorm(4)
1  1  1.0000000000000000  1.0000000000000000  1.0000000000000000  1.0000000000000000  1.0000000000000000
1  2  0.0028264970995254  0.0018828719443888  0.0014289099428535  0.0039439618931841  0.0021543142030861
1  3  0.0000019852262002  0.0000009146078748  0.0000087277900904  0.0000057307160882  0.0000051453312708
1  4  0.0000000299145789  0.0000000089835596  0.0000000215954161  0.0000000480469191  0.0000000090492238
```

`avgConvRate(0) `, `avgConvRate(1)`, ... and `resNorm(0)`, `resNorm(1)`, ... come in the same order as in the preCICE configuration specified. It would be much easier to understand, however, if we directly specified the names: `avgConvRate(Forces)` or `resNorm(Displacements)`.

While doing this, we could also bring some language consistency into these files (`ResNorm`, `TotalIterations` ...) 

... and also add the names to the convergence INFO statement:
```
preCICE: relative convergence measure: two-norm diff = 1.79666e-09, relative limit = 1.92741e-10, conv = false
preCICE: relative convergence measure: two-norm diff = 4.39099e-09, relative limit = 2.07417e-10, conv = false
preCICE: relative convergence measure: two-norm diff = 0.0336136, relative limit = 2.06253e-05, conv = false
preCICE: relative convergence measure: two-norm diff = 0.0114555, relative limit = 2.06612e-05, conv = false
preCICE: relative convergence measure: two-norm diff = 0.00739805, relative limit = 2.07122e-05, conv = false
```

### PR DESCRIPTION
**Pull request by DavidSCN**
## Main changes of this PR
...as discussed and closes #555 

## Author's checklist

* [x] I added a changelog file with this PR number in `docs/changelog/` if there are noteworthy changes.
* [x] I ran `tools/formatting/check-format` and everything is formatted correctly.
* [x] I sticked to C++14 features.
* [x] I sticked to CMake version 3.10.
* [ ] I squashed / am about to squash all commits that should be seen as one.

## Reviewers' checklist

* [x] Does the changelog entry make sense? Is it formatted correctly?
* [x] Do you understand the code changes?
* [ ] (more questions/tasks)

### CODE REVIEW

File: src/cplscheme/impl/MinIterationConvergenceMeasure.hpp [hunk 1]
```diff
@@ -35,10 +35,10 @@ class MinIterationConvergenceMeasure : public ConvergenceMeasure {
     return _isConvergence;
   }
 
-  virtual std::string printState()
+  virtual std::string printState(const std::string &dataName)
   {
     std::ostringstream os;
-    os << "min iteration convergence measure: ";
+    os << "min iteration convergence measure of data \"" << dataName << "\": ";
```
Comment (uekerman) [inline src/cplscheme/impl/MinIterationConvergenceMeasure.hpp:41]: I guess it's debatable whether we want it here. Of course, it's already weird / wrong that a min-iteration convergence measure requires a data field :see_no_evil: . But it does not get better if we output it.

**Review by uekerman — APPROVED** [review]
Looks good, only a minor comment below.
I guess you tested this with some example, right?
I did not.
And I would say it's worth a changelog entry, it does alter user experience at least a little bit.

File: src/cplscheme/impl/MinIterationConvergenceMeasure.hpp (hunk 1 above)
Comment (DavidSCN) [reply src/cplscheme/impl/MinIterationConvergenceMeasure.hpp]: I removed it

**Comment by DavidSCN** [comment]
I tested a single test case, yes. Only inconsistency: do we want to use quotation marks for the names `"Displacement"` or not. We have now
```
Mapping Force conservative...
```
and
```
relative two-norm diff of data "Force":
```

**Comment by uekerman** [comment]
I would use quotation marks here. We also have them in all our error messages. So, the mapping one is the inconstent one.

> I tested a single test case, yes.

:+1:

**Comment by DavidSCN** [comment]
Added it for the RBF mapping (which is the only one required AFAI see).

**Comment by fsimonis** [comment]
@uekerman 

Output of `elastic-tube-1d/solid-cpp`:
```
(0) 16:45:16 [impl::SolverInterfaceImpl]:434 in advance: iteration: 9 of 40, time-window: 100, time: 0.99 of 1, time-window-size: 0.01, max-timestep-length: 0.01, ongoing: yes, time-window-complete: no, read-iteration-checkpoint
(0) 16:45:16 [cplscheme::BaseCouplingScheme]:513 in measureConvergence: relative convergence measure: relative two-norm diff of data "Pressure" = 5.13109e-09, limit = 1e-09, normalization = 1829.98, conv = false
(0) 16:45:16 [cplscheme::BaseCouplingScheme]:513 in measureConvergence: relative convergence measure: relative two-norm diff of data "CrossSectionLength" = 3.46562e-12, limit = 1e-09, normalization = 9.99261, conv = true
(0) 16:45:16 [impl::SolverInterfaceImpl]:434 in advance: iteration: 10 of 40, time-window: 100, time: 0.99 of 1, time-window-size: 0.01, max-timestep-length: 0.01, ongoing: yes, time-window-complete: no, read-iteration-checkpoint
(0) 16:45:16 [cplscheme::BaseCouplingScheme]:513 in measureConvergence: relative convergence measure: relative two-norm diff of data "Pressure" = 4.26289e-10, limit = 1e-09, normalization = 1829.98, conv = true
(0) 16:45:16 [cplscheme::BaseCouplingScheme]:513 in measureConvergence: relative convergence measure: relative two-norm diff of data "CrossSectionLength" = 5.00318e-14, limit = 1e-09, normalization = 9.99261, conv = true
```

**Pull request merged**